internal/component/loki/secretfilter/secretfilter.go

```diff
@@ -154,10 +155,12 @@ func (c *Component) Run(ctx context.Context) error {
 			for _, f := range c.fanout {
```

Comment:
nit: I think that technically you can release the lock before the for loop because the processing is already done. This way you release a little bit earlier and you only have one release call. Not very important though, feel free to ignore

Reply:
I think I did this based on the behaviour of other components I saw, where they kept the lock while iterating over the fanouts (in case that would be changed by the config update)